File: scripts/remediate_db.py
import json
import psycopg2
import os

# Database connection from environment
# Note: In production use env vars, here we use the values from .env directly for the script
db_config = {
    "host": "localhost",
    "port": 5432,
    "database": "ai_store_manager",
    "user": "postgres",
    "password": "101914"
}

def remediate_database():
    print("--- Remediating Database Categories ---")
    
    # Load the corrected real_products mapping
    real_products_path = 'd:/AI Store Manger/data/real_products.json'
    if not os.path.exists(real_products_path):
        print("real_products.json not found. Run fetch_real_data.py first.")
        return

    with open(real_products_path, 'r', encoding='utf-8') as f:
        real_products = json.load(f)

    # Create a mapping of product name -> category
    name_to_cat = {p['name']: p['category'] for p in real_products}

    try:
        conn = psycopg2.connect(**db_config)
        cur = conn.cursor()

        # Get all demo-store items
        cur.execute("SELECT store_item_id, normalized_product_name, master_category_name FROM store_sku_registry WHERE store_id = 'demo-store'")
        rows = cur.fetchall()

        updates = 0
        for sku_id, name, current_cat in rows:
            if name in name_to_cat:
                new_cat = name_to_cat[name]
                if new_cat != current_cat:
                    cur.execute(
                        "UPDATE store_sku_registry SET master_category_name = %s WHERE store_id = 'demo-store' AND store_item_id = %s",
                        (new_cat, sku_id)
                    )
                    updates += 1

        conn.commit()
        print(f"Successfully updated {updates} items in store_sku_registry.")
        
        cur.close()
        conn.close()
    except Exception as e:
        print(f"Database error: {e}")

if __name__ == "__main__":
    remediate_database()

File: scripts/remediate_db.js
const { Pool } = require('pg');
const fs = require('fs');
const path = require('path');

async function remediate() {
    console.log("--- Remediating Database Categories (Node.js) ---");

    // Config from .env or manual for this task
    const pool = new Pool({
        user: 'postgres',
        host: 'localhost',
        database: 'ai_store_manager',
        password: '101914',
        port: 5432,
    });

    const productsPath = path.join(__dirname, '..', 'data', 'real_products.json');
    if (!fs.existsSync(productsPath)) {
        console.log("real_products.json not found.");
        return;
    }

    const realProducts = JSON.parse(fs.readFileSync(productsPath, 'utf8'));
    const nameToCat = {};
    realProducts.forEach(p => {
        nameToCat[p.name] = p.category;
    });

    try {
        const client = await pool.connect();
        const res = await client.query("SELECT store_item_id, normalized_product_name, master_category_name FROM store_sku_registry WHERE store_id = 'demo-store'");

        let updates = 0;
        for (const row of res.rows) {
            const newCat = nameToCat[row.normalized_product_name];
            if (newCat && newCat !== row.master_category_name) {
                await client.query(
                    "UPDATE store_sku_registry SET master_category_name = $1 WHERE store_id = 'demo-store' AND store_item_id = $2",
                    [newCat, row.store_item_id]
                );
                updates++;
            }
        }

        console.log(`Successfully updated ${updates} items.`);
        client.release();
    } catch (err) {
        console.error("DB Error:", err);
    } finally {
        await pool.end();
    }
}

remediate();
